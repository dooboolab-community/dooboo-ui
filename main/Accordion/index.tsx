import {Animated, StyleProp, ViewStyle} from 'react-native';
import React, {FC, useRef} from 'react';

import AccordionItem from './AccordionItem';
import styled from '@emotion/native';

const Container = styled.View`
  flex-direction: column;
  align-items: center;
`;

export interface Datum {
  title: string;
  bodies: string[];
}

export type Styles = {
  titleContainer?: StyleProp<ViewStyle>;
  bodyContainer?: StyleProp<ViewStyle>;
};

interface Props {
  style?: ViewStyle;
  styles?: Styles;
  data: Datum[];
  shouldAnimate?: boolean;
  collapseOnStart?: boolean;
  animDuration?: number;
  activeOpacity?: number;
  toggleElement?: React.ReactElement;
  renderTitle?: (item: string) => React.ReactElement;
  renderBody?: (item: string) => React.ReactElement;
}

const Accordion: FC<Props> = (props) => {
  const {
    style,
    styles,
    data,
    collapseOnStart,
    shouldAnimate,
    animDuration,
    activeOpacity,
    toggleElement,
    renderTitle,
    renderBody,
  } = props;

  const dropDownAnimValueList = useRef(
    data.map(() => new Animated.Value(0)),
  ).current;

  return (
    <Container style={style}>
      {data.map((datum, titleKey) => {
        return (
          <AccordionItem
            key={titleKey}
            testID={`${titleKey}`}
            styles={styles}
            datum={datum}
            collapseOnStart={collapseOnStart}
            shouldAnimate={shouldAnimate}
            animDuration={animDuration}
            activeOpacity={activeOpacity}
            dropDownAnimValueList={dropDownAnimValueList[titleKey]}
            sumOfPrecedingTranslateY={dropDownAnimValueList
              .filter((item, idx) => idx < titleKey)
              .map((value) => ({translateY: value}))}
            toggleElement={toggleElement}
            renderTitle={renderTitle}
            renderBody={renderBody}
          />
        );
      })}
    </Container>
  );
};

export {Accordion};
