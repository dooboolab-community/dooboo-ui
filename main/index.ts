export * from './uis/Typography';
export * from './uis/RadioGroup';
export * from './modals/Snackbar';
export * from './uis/Icon';
export * from './uis/Hr';
export * from './uis/Button';
export * from './uis/ButtonGroup';
export * from './uis/IconButton';
export * from './uis/Checkbox';
export * from './uis/EditText';
export * from './uis/NetworkImage';
export * from './uis/ProgressBar';
export * from './uis/ProgressCircle';
export * from './uis/LoadingIndicator';
export * from './uis/Accordion';
export * from './uis/StatusbarBrightness';
export * from './uis/SwitchToggle';
export * from './uis/Fabs';
export * from './uis/Rating';
export * from './uis/CustomPressable';
export * from './providers';
