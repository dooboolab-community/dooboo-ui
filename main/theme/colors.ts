import {Theme} from '@emotion/react';

export type ThemeType = 'light' | 'dark';

export const colors = {
  black: 'black',
  white: 'white',
  charcoalGray: '#414141',
  gray: '#6D6D6D',
  cloud: '#C4C4C4',
  lightGray: '#EDEDED',
  brightTurquoise: '#00D9D5',
  ruddy: '#FF002E',
  yellow: '#FFEB14',
  blueberry: '#5398FF',
  harlequin: '#33FF2F',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
  negative: '#ff7676',
  positive: '#00BA90',
};

export type Colors = typeof colors;

export const light = {
  primary: colors.black,
  paper: colors.lightGray,
  disabled: colors.cloud,
  background: colors.white,
  secondary: colors.brightTurquoise,
  danger: colors.ruddy,
  warning: colors.yellow,
  light: colors.lightGray,
  info: colors.blueberry,
  success: colors.harlequin,
  textPrimary: colors.black,
  text: colors.black,
  placeholder: colors.gray,
  textDisabled: colors.cloud,
  textContrast: colors.white,
  textValidation: colors.ruddy,
};

export type DoobooTheme = typeof light & {
  isPortrait?: boolean;
  isDesktop?: boolean;
  isTablet?: boolean;
  isMobile?: boolean;
};

export const dark = {
  primary: colors.white,
  paper: colors.charcoalGray,
  disabled: colors.charcoalGray,
  background: colors.black,
  secondary: colors.brightTurquoise,
  danger: colors.ruddy,
  warning: colors.yellow,
  light: colors.charcoalGray,
  info: colors.blueberry,
  success: colors.harlequin,
  textPrimary: colors.white,
  text: colors.white,
  placeholder: colors.gray,
  textDisabled: colors.charcoalGray,
  textContrast: colors.black,
  textValidation: colors.ruddy,
};

export const theme = {
  light,
  dark,
};

export interface ThemeParam {
  light: Partial<Theme>;
  dark: Partial<Theme>;
}

export const createDoobooTheme = ({
  type,
  themes,
}: {
  type?: ThemeType;
  themes?: ThemeParam;
}): Partial<Theme> => {
  switch (type) {
    case 'dark':
      return {...theme.dark, ...themes?.dark};
    case 'light':
    default:
      return {...theme.light, ...themes?.light};
  }
};
