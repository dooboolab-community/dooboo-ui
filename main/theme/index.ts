export {ThemeProvider, useTheme, withTheme} from './ThemeProvider';
export {DoobooTheme, ThemeType, dark, light} from './colors';
