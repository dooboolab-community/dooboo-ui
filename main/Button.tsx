import {
  ActivityIndicator,
  LayoutRectangle,
  Platform,
  TouchableOpacity,
} from 'react-native';
import {DoobooTheme, light, withTheme} from './theme';
import React, {useRef, useState} from 'react';
import type {
  StyleProp,
  TextProps,
  TextStyle,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';

import type {FC} from 'react';
import styled from '@emotion/native';
import {useHover} from 'react-native-web-hooks';

type Styles = {
  container?: StyleProp<ViewStyle>;
  text?: StyleProp<TextStyle>;
  disabledButton?: StyleProp<ViewStyle>;
  disabledText?: StyleProp<TextStyle>;
  hovered?: StyleProp<ViewStyle>;
};

const Container = styled.View`
  align-self: stretch;
  padding: 8px 12px;
  background-color: ${({theme}) => theme.primary || light.primary};

  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Text = styled.Text`
  font-size: 14px;
  color: white;
`;

export interface ButtonProps {
  testID?: string;
  indicatorColor?: string;
  loading?: boolean;
  disabled?: boolean;
  style?: StyleProp<ViewStyle>;
  styles?: Styles;
  leftElement?: React.ReactElement;
  rightElement?: React.ReactElement;
  activeOpacity?: TouchableOpacityProps['activeOpacity'];
  text?: string;
  onPress?: TouchableOpacityProps['onPress'];
  touchableOpacityProps?: Partial<TouchableOpacityProps>;
  textProps?: Partial<TextProps>;
}

const StyledButton: FC<ButtonProps & {theme: DoobooTheme}> = ({
  testID,
  theme,
  disabled,
  loading,
  style,
  styles,
  indicatorColor = theme.disabled,
  leftElement,
  rightElement,
  activeOpacity = 0.7,
  text,
  onPress,
  touchableOpacityProps,
  textProps,
}) => {
  const ref = useRef<TouchableOpacity>(null);
  const hovered = useHover(ref);
  const [layout, setLayout] = useState<LayoutRectangle>();

  const compositeStyles: Styles = {
    disabledButton: {
      backgroundColor: theme.primary,
      borderColor: theme.primary,
    },
    disabledText: {
      color: theme.primary,
    },
    hovered: {
      borderColor: theme.primary,
      shadowColor: 'black',
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowOpacity: 0.24,
      shadowRadius: 16,
      elevation: 10,
    },
    ...styles,
  };

  return (
    <TouchableOpacity
      testID={testID}
      ref={Platform.select({
        web: ref,
        default: undefined,
      })}
      activeOpacity={activeOpacity}
      onPress={onPress}
      delayPressIn={50}
      disabled={disabled}
      style={style}
      {...touchableOpacityProps}>
      {loading ? (
        <Container
          testID="loading-view"
          style={[
            compositeStyles.container,
            {
              width: layout?.width,
              height: layout?.height,
            },
            hovered && !disabled && compositeStyles.hovered,
            disabled && compositeStyles.disabledButton,
          ]}>
          <ActivityIndicator size="small" color={indicatorColor} />
        </Container>
      ) : (
        <Container
          testID="button-view"
          style={[
            compositeStyles.container,
            hovered && !disabled && compositeStyles.hovered,
            disabled && compositeStyles.disabledButton,
          ]}
          onLayout={(e) => setLayout(e.nativeEvent.layout)}>
          {leftElement}
          <Text
            style={[
              compositeStyles.text,
              disabled && compositeStyles.disabledText,
            ]}
            {...textProps}>
            {text}
          </Text>
          {rightElement}
        </Container>
      )}
    </TouchableOpacity>
  );
};

StyledButton.defaultProps = {theme: light};

export const Button = withTheme(StyledButton);
