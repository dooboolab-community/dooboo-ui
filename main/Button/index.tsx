import {
  ActivityIndicator,
  LayoutRectangle,
  Platform,
  TouchableOpacity,
} from 'react-native';
import {
  ButtonText,
  ButtonType,
  ButtonWrapper,
} from '../Styled/StyledComponents';
import {DoobooTheme, light, withTheme} from '../theme';
import React, {useRef, useState} from 'react';
import type {
  StyleProp,
  TextProps,
  TextStyle,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';
import styled, {css} from '@emotion/native';

import type {FC} from 'react';
import {useHover} from 'react-native-web-hooks';

type Styles = {
  container?: StyleProp<ViewStyle>;
  text?: StyleProp<TextStyle>;
  disabledButton?: StyleProp<ViewStyle>;
  disabledText?: StyleProp<TextStyle>;
  hovered?: StyleProp<ViewStyle>;
};

type ButtonSize = 'small' | 'medium' | 'large';

const ButtonContainer = styled(ButtonWrapper)<{
  type: ButtonType;
  size?: ButtonSize;
  outlined?: boolean;
  disabled?: boolean;
}>`
  align-self: stretch;
  padding: ${({size}) =>
    size === 'large'
      ? '14px 24px'
      : size === 'small'
      ? '4px 12px'
      : '10px 20px'};
  border-radius: ${({size}) => (size === 'large' ? '24px' : '20px')};

  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

export interface ButtonProps {
  testID?: string;
  indicatorColor?: string;
  loading?: boolean;
  disabled?: boolean;
  outlined?: boolean;
  style?: StyleProp<ViewStyle>;
  styles?: Styles;
  leftElement?: React.ReactElement;
  rightElement?: React.ReactElement;
  activeOpacity?: TouchableOpacityProps['activeOpacity'];
  text?: string;
  onPress?: TouchableOpacityProps['onPress'];
  touchableOpacityProps?: Partial<TouchableOpacityProps>;
  textProps?: Partial<TextProps>;
  type?: ButtonType;
  size?: ButtonSize;
}

const ButtonComponent: FC<ButtonProps & {theme: DoobooTheme}> = ({
  testID,
  theme,
  disabled,
  loading,
  style,
  styles,
  indicatorColor = theme.text,
  leftElement,
  rightElement,
  activeOpacity = 0.6,
  text,
  onPress,
  touchableOpacityProps,
  textProps,
  type = 'primary',
  size,
  outlined = false,
}) => {
  const ref = useRef<TouchableOpacity>(null);
  const hovered = useHover(ref);
  const [layout, setLayout] = useState<LayoutRectangle>();

  const compositeStyles: Styles = {
    disabledButton: css`
      background-color: ${!outlined ? theme.disabled : theme.background};
      border-color: ${theme.disabled};
    `,
    disabledText: {
      color: !outlined ? theme.background : theme.textDisabled,
    },
    hovered: {
      shadowColor: 'black',
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowOpacity: 0.24,
      shadowRadius: 16,
      elevation: 10,
    },
    ...styles,
  };

  return (
    <TouchableOpacity
      testID={testID}
      ref={Platform.select({
        web: ref,
        default: undefined,
      })}
      activeOpacity={activeOpacity}
      onPress={onPress}
      delayPressIn={50}
      disabled={disabled || loading}
      style={style}
      {...touchableOpacityProps}>
      {loading ? (
        <ButtonContainer
          testID="loading-view"
          disabled
          style={[
            compositeStyles.container,
            {
              width: layout?.width,
              height: layout?.height,
            },
            hovered && !disabled && compositeStyles.hovered,
            disabled && compositeStyles.disabledButton,
          ]}
          type={type}
          size={size}
          outlined={outlined}>
          <ActivityIndicator size="small" color={indicatorColor} />
        </ButtonContainer>
      ) : (
        <ButtonContainer
          testID="button-container"
          style={[
            compositeStyles.container,
            hovered && !disabled && compositeStyles.hovered,
            disabled && compositeStyles.disabledButton,
          ]}
          onLayout={(e) => setLayout(e.nativeEvent.layout)}
          type={type}
          size={size}
          disabled={disabled}
          outlined={outlined}>
          {leftElement}
          <ButtonText
            outlined={outlined}
            type={type}
            style={[
              compositeStyles.text,
              disabled && compositeStyles.disabledText,
            ]}
            {...textProps}>
            {text}
          </ButtonText>
          {rightElement}
        </ButtonContainer>
      )}
    </TouchableOpacity>
  );
};

ButtonComponent.defaultProps = {theme: light};

export const Button = withTheme(ButtonComponent);
