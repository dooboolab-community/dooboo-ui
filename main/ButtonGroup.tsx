import {DoobooTheme, light, useTheme, withTheme} from './theme';
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

interface Styles {
  container: StyleProp<ViewStyle>;
  button: ViewStyle;
  selectedButton: ViewStyle;
  text: StyleProp<TextStyle>;
  selectedText: StyleProp<TextStyle>;
}

interface Props<T> {
  testID?: string;
  theme: DoobooTheme;
  borderRadius?: number;
  borderWidth?: number;
  style?: StyleProp<ViewStyle>;
  styles?: Styles;
  data: T[];
  color?: string;
  onPress?: (i: number) => void;
  selected?: number;
}

function StyledButtonGroup<T>(props: Props<T>): React.ReactElement {
  const {theme} = useTheme();

  const {
    borderRadius = 0,
    selected = 0,
    borderWidth = 1,
    color = theme.text,
    testID,
    style,
    data,
    onPress,
    styles,
  } = props;

  return (
    <View
      testID={testID}
      style={StyleSheet.flatten([
        {borderColor: color},
        styles?.container,
        style,
      ])}>
      {data.map((text, i) => {
        return (
          <TouchableOpacity
            key={i}
            testID={`CHILD_${i}`}
            activeOpacity={0.85}
            style={{flex: 1}}
            onPress={(): void => {
              if (onPress) onPress(i);
            }}>
            <View
              style={StyleSheet.flatten([
                selected === i
                  ? {...styles?.selectedButton, backgroundColor: color}
                  : {...styles?.button, borderColor: color},
                i === 0
                  ? {
                      borderLeftWidth: borderWidth,
                      borderTopWidth: borderWidth,
                      borderBottomWidth: borderWidth,
                      borderTopLeftRadius: borderRadius,
                      borderBottomLeftRadius: borderRadius,
                    }
                  : i === data.length - 1
                  ? {
                      borderRightWidth: borderWidth,
                      borderLeftWidth: borderWidth,
                      borderTopWidth: borderWidth,
                      borderBottomWidth: borderWidth,
                      borderBottomRightRadius: borderRadius,
                      borderTopRightRadius: borderRadius,
                    }
                  : {
                      borderLeftWidth: borderWidth,
                      borderTopWidth: borderWidth,
                      borderBottomWidth: borderWidth,
                    },
                {borderColor: color},
              ])}>
              <Text
                style={
                  selected === i
                    ? [styles?.selectedText, {color: theme.textContrast}]
                    : [styles?.text, {color: theme.text}]
                }>
                {text}
              </Text>
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
}

StyledButtonGroup.defaultProps = {
  theme: light,
  styles: {
    container: {
      backgroundColor: 'transparent',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      alignSelf: 'stretch',
      minHeight: 40,
      marginTop: 24,
    },
    button: {
      alignSelf: 'stretch',
      minHeight: 40,

      justifyContent: 'center',
      alignItems: 'center',
    },
    selectedButton: {
      alignSelf: 'stretch',
      minHeight: 40,

      justifyContent: 'center',
      alignItems: 'center',
    },
    text: {
      fontSize: 14,
      textAlign: 'center',
      alignSelf: 'center',
    },
    selectedText: {
      fontSize: 14,
      textAlign: 'center',
      alignSelf: 'center',
    },
  },
  data: ['option 1', 'option 2'],
};

export const ButtonGroup = withTheme<any>(StyledButtonGroup);
