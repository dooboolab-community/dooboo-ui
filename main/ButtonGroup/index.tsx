import React from 'react';
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

import {useTheme} from '@dooboo-ui/theme';

interface Styles {
  container: StyleProp<ViewStyle>;
  button: ViewStyle;
  selectedButton: ViewStyle;
  text: StyleProp<TextStyle>;
  selectedText: StyleProp<TextStyle>;
}

interface Props<T> {
  testID?: string;
  borderRadius?: number;
  borderWidth?: number;
  style?: StyleProp<ViewStyle>;
  styles?: Styles;
  data: T[];
  color?: string;
  onPress?: (i: number) => void;
  selectedIndex?: number;
}

export function ButtonGroup<T>(props: Props<T>): React.ReactElement {
  const {theme} = useTheme();

  const {
    borderRadius = 0,
    selectedIndex = 0,
    borderWidth = 1,
    color = theme.text,
    testID,
    style,
    data,
    onPress,
    styles,
  } = props;

  const borderWidthAndRadius = (index: number): object => {
    const fullWidthAndRadius = {
      borderLeftWidth: borderWidth,
      borderRightWidth: borderWidth,
      borderTopWidth: borderWidth,
      borderBottomWidth: borderWidth,

      borderTopLeftRadius: borderRadius,
      borderBottomLeftRadius: borderRadius,
      borderTopRightRadius: borderRadius,
      borderBottomRightRadius: borderRadius,
    };

    const isFirst = index === 0;
    const isLast = index === data.length - 1;

    const borderForFirstElement = {
      ...fullWidthAndRadius,
      borderTopRightRadius: undefined,
      borderBottomRightRadius: undefined,
    };

    const borderForLastElement = {
      ...fullWidthAndRadius,
      borderTopLeftRadius: undefined,
      borderBottomLeftRadius: undefined,
    };

    const borderForMiddleElement = {
      borderRightWidth: borderWidth,
      borderTopWidth: borderWidth,
      borderBottomWidth: borderWidth,
    };

    if (data.length === 1) return fullWidthAndRadius;

    if (isFirst) return borderForFirstElement;

    if (isLast) {
      if (data.length === 2)
        return {
          ...borderForLastElement,
          borderLeftWidth: undefined,
        };

      return borderForLastElement;
    }

    return borderForMiddleElement;
  };

  return (
    <View
      testID={testID}
      style={StyleSheet.flatten([
        {borderColor: color},
        styles?.container,
        style,
      ])}
    >
      {data.map((text, i) => {
        return (
          <TouchableOpacity
            key={i}
            activeOpacity={0.85}
            style={{flex: 1}}
            onPress={(): void => {
              if (onPress) onPress(i);
            }}
          >
            <View
              testID={`CHILD_${i}`}
              style={StyleSheet.flatten([
                selectedIndex === i
                  ? {...styles?.selectedButton, backgroundColor: color}
                  : {...styles?.button, borderColor: color},
                borderWidthAndRadius(i),
                {borderColor: color},
              ])}
            >
              <Text
                style={
                  selectedIndex === i
                    ? [styles?.selectedText, {color: theme.textContrast}]
                    : [styles?.text, {color: theme.text}]
                }
              >
                {text}
              </Text>
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
}

ButtonGroup.defaultProps = {
  styles: {
    container: {
      backgroundColor: 'transparent',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      alignSelf: 'stretch',
      minHeight: 40,
      marginTop: 24,
    },
    button: {
      alignSelf: 'stretch',
      minHeight: 40,

      justifyContent: 'center',
      alignItems: 'center',
    },
    selectedButton: {
      alignSelf: 'stretch',
      minHeight: 40,

      justifyContent: 'center',
      alignItems: 'center',
    },
    text: {
      fontSize: 14,
      textAlign: 'center',
      alignSelf: 'center',
    },
    selectedText: {
      fontSize: 14,
      textAlign: 'center',
      alignSelf: 'center',
    },
  },
  data: ['option 1', 'option 2'],
};
