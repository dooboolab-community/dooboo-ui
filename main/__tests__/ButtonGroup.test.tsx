import {fireEvent, render, RenderAPI} from '@testing-library/react-native';
import {createComponent, createTestProps} from '../../test/testUtils';

import {ButtonGroup} from '../../main';
import {theme} from '../theme/colors';
import {ThemeProvider} from '../theme';

describe('[ButtonGroup]', () => {
  const themeType = 'light';

  const handlePress = jest.fn();

  const renderButtonGroup = (): RenderAPI =>
    render(
      <ThemeProvider initialThemeType={themeType}>
        <ButtonGroup
          initialIndex={0}
          data={['option 1', 'option 2']}
          onPress={handlePress}
        />
      </ThemeProvider>,
    );

  beforeEach(() => {
    handlePress.mockClear();
  });

  it('renders without crashing', () => {
    const props = createTestProps();

    const testingLib = render(createComponent(<ButtonGroup {...props} />));

    expect(testingLib.toJSON()).toMatchSnapshot();
  });

  it('listens to press event', () => {
    const {getByText} = renderButtonGroup();

    fireEvent.press(getByText('option 1'));

    expect(handlePress).toBeCalled();
  });

  it('selects option on press', () => {
    const {getByText} = renderButtonGroup();

    expect(getByText('option 1')).toHaveStyle({
      color: theme[themeType].textContrast,
    });

    fireEvent.press(getByText('option 2'));

    expect(getByText('option 1')).not.toHaveStyle({
      color: theme[themeType].textContrast,
    });
  });

  it('distinguishes between selected and unselected options.', () => {
    const {getByText} = renderButtonGroup();

    expect(getByText('option 1')).toHaveStyle({
      color: theme[themeType].textContrast,
    });

    expect(getByText('option 2')).toHaveStyle({
      color: theme[themeType].text,
    });
  });
});
