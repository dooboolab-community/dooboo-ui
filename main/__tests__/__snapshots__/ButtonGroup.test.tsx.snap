// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ButtonGroup] renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "transparent",
      "borderColor": "black",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 24,
      "minHeight": 40,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="CHILD_0"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "black",
          "borderBottomLeftRadius": 0,
          "borderBottomWidth": 1,
          "borderColor": "black",
          "borderLeftWidth": 1,
          "borderTopLeftRadius": 0,
          "borderTopWidth": 1,
          "justifyContent": "center",
          "minHeight": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        option 1
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="CHILD_1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderBottomRightRadius": 0,
          "borderBottomWidth": 1,
          "borderColor": "black",
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopRightRadius": 0,
          "borderTopWidth": 1,
          "justifyContent": "center",
          "minHeight": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "color": "black",
            },
          ]
        }
      >
        option 2
      </Text>
    </View>
  </View>
</View>
`;
